src/components/Header/Header.tsx

```diff
@@ -45,35 +54,33 @@ class Header extends UIComponent<ReactProps<HeaderProps>, any> {
     ...commonPropTypes.createCommon({ color: true }),
     description: customPropTypes.itemShorthand,
     textAlign: PropTypes.oneOf(['left', 'center', 'right', 'justified']),
+    rtlAttributes: PropTypes.func,
   }
 
   static defaultProps = {
     as: 'h1',
+    rtlAttributes: headerRtlAttributes,
   }
 
   static Description = HeaderDescription
 
-  renderComponent({ ElementType, classes, variables: v, unhandledProps }) {
-    const { children, content, description } = this.props
+  renderComponent({ ElementType, classes, variables: v, unhandledProps, rtlAttributes }) {
+    const { children, description, content } = this.props
 
-    if (childrenExist(children)) {
-      return (
-        <ElementType {...unhandledProps} className={classes.root}>
-          {children}
-        </ElementType>
-      )
-    }
+    const hasChildren = childrenExist(children)
+    const contentElement = childrenExist(children) ? children : content
 
     return (
-      <ElementType {...unhandledProps} className={classes.root}>
-        {content}
-        {HeaderDescription.create(description, {
-          defaultProps: {
-            variables: {
-              ...(v.descriptionColor && { color: v.descriptionColor }),
+      <ElementType {...rtlAttributes.root} {...unhandledProps} className={classes.root}>
+        {description ? contentElement : addRtlSupport(contentElement)}
```

Comment:
don't like the fact that this general approach like 'calculate attributes during rendering and apply them to <ElementType />' breaks here: the `addRtlSupport` logic will unconditionally apply the same 'rtl behavior' (children-dependent logic) to `contentElement`, whatever this is. 

Could we generalize our approach to handle this case as well? Like, for example, introduce `description` prop for RTL attributes provider? Otherwise, I am not sure how this approach might scale to handle similar cases

Reply:
The general problem is that, in cases when the calculation for the rtl props are failing (we are returning empty object) we want to add `<span dir='auto'>` around the element. I am open to other ideas..